src/bridger/Bridger.sol

```diff
@@ -308,9 +310,9 @@ contract Bridger is
         amountBought = ERC20(wstETH).balanceOf(address(this)) - balanceBefore;
     }
 
-    function _stakeAssets(address _asset, uint256 _amount) private {
+    function _stakeAssets(address _asset, uint256 _amount) private returns (uint256) {
         IERC20(_asset).approve(address(sUSDe), _amount);
-        IsUSDe(sUSDe).deposit(_amount, address(this));
+        return IsUSDe(sUSDe).deposit(_amount, address(this));
```

Comment:
I dont think this will work, because it will accumulate the amount on each deposit.

I recommend just calling balanceOf before and after and using the delta

Reply:
or maybe better to multiply the amount staked for the exchange rate, the contract may have it

Reply:
>  because it will accumulate the amount on each deposit

AFAIK, it does not accumulate, it just returns the value of shared that you got by depositing a given amount

> or maybe better to multiply the amount staked for the exchange rate, the contract may have it

`deposit` calls `previewDeposit` which returns the amount of shares by calling `_convertToShares` which kind of does the multiplication you are suggesting

Reply:
perfect, sounds good